src/pages/Contests/Contests.js

```diff
@@ -0,0 +1,307 @@
+    queryDefault.divs
+      ? {
+          values: [
+            queryDefault.divs.includes('Div. 1'),
+            queryDefault.divs.includes('Div. 2'),
+            queryDefault.divs.includes('Div. 3'),
+            queryDefault.divs.includes('Div. 4'),
+            queryDefault.divs.includes('Educational'),
+            queryDefault.divs.includes('Global'),
+          ],
+        }
+      : { values: [false, false, false, false, false, false] }
+  )
+
+  //available difficulty filters
+  const difficultyFilters = [
+    'Div. 1',
+    'Div. 2',
+    'Div. 3',
+    'Div. 4',
+    'Educational',
+    'Global',
+  ]
+
+  //FIlter parameters
+  const [gym, setGym] = useState(queryDefault.gym ? queryDefault.gym : false)
+  const [gymCount, setGymCount] = useState(queryDefault.gym ? true : false)
+  const [mentorr, setMentorr] = useState(
+    queryDefault.mentor ? queryDefault.mentor : false
+  )
+  const [mentorrCount, setMentorrCount] = useState(
+    queryDefault.mentor ? true : false
+  )
+
+  const [difficultyQueries, setDifficultyQueries] = useState(
+    queryDefault.divs ? queryDefault.divs.split(',') : []
+  )
+
+  const [isDiffChange, setIsDiffChange] = useState(
+    queryDefault.divs ? true : false
+  )
+
+  //Function to change the difficulty
+  const changeDifficultyFilter = (event, lev) => {
+    const res = event.target.checked
+    setIsDiffChange(true)
+    const difficultyAdd = difficultyLevels[lev]
+    if (res) {
+      setDifficultyQueries([...difficultyQueries, difficultyAdd])
+      console.log(difficultyQueries)
+      setDisplayDiff(
+        update(displayDiff, {
+          values: {
+            [lev]: {
+              $set: true,
+            },
+          },
+        })
+      )
+    } else {
+      const newList = difficultyQueries.filter(
+        (item) => item != difficultyFilters[lev]
+      )
+      setDifficultyQueries(newList)
+      setDisplayDiff(
+        update(displayDiff, {
+          values: {
+            [lev]: {
+              $set: false,
+            },
+          },
+        })
+      )
+    }
+  }
+  const handleSubmit = (e) => {
+    e.preventDefault()
+
+    var urlTo = '/contests?'
+
+    if (isDiffChange && difficultyQueries.length > 0) {
+      urlTo =
+        urlTo +
+        `divs=${JSON.stringify(difficultyQueries)
+          .replace(/"/g, '')
+          .replace(/]|[[]/g, '')}`
+    }
+    console.log(urlTo)
+
+    if (gymCount) {
+      urlTo = urlTo + `;gym=${JSON.stringify(gym)}`
+    }
+
+    console.log(urlTo)
+
+    if (mentorrCount) {
+      urlTo = urlTo + `;mentor=${JSON.stringify(mentorr)}`
+    }
+    console.log(urlTo)
+
+    window.location.href = urlTo
+  }
+
+  //searching for a contest
+  const handleSearch = (e) => {
+    // e.preventDefault()
+    const searchUrl = `/contests/?divs=${searchText}`
+    window.location.href = searchUrl
+  }
+
+  //opening filter block
+  function openNav() {
+    document.getElementById('mySidenav1').classList.remove('sideNavClose')
```

Comment:
Try to find a way to not use this vanilla DOM manipulation as this is not a good practice, Try finding some alternate method.